pkg/healthcheck/healthcheck.go

```diff
@@ -571,7 +592,9 @@ func validateControlPlanePods(pods []v1.Pod) error {
 
 	for _, pod := range pods {
 		if pod.Status.Phase == v1.PodRunning {
-			name := strings.Split(pod.Name, "-")[0]
+			// strip the single-namespace "linkerd-" prefix if it exists
+			name := strings.TrimPrefix(pod.Name, "linkerd-")
```

Comment:
Consider making `linkerd-` a constant and re-using in `cli/cmd/install.go`?

Related to previous comment, I'd really prefer to do away with this special-casing, as it will increase our support burden.